toolchain/check/eval_inst.cpp

```diff
@@ -197,6 +222,24 @@ auto EvalConstantInst(Context& /*context*/, SemIRLoc /*loc*/,
       .type_id = inst.type_id, .elements_id = SemIR::InstBlockId::Empty});
 }
 
+auto EvalConstantInst(Context& context, SemIRLoc loc,
+                      SemIR::ImplSymbolicWitness inst) -> ConstantEvalResult {
+  auto result = EvalLookupSingleImplWitness(
+      context, UnwrapSemIRLoc()(context, loc), inst);
+  if (!result.has_value()) {
+    // We use NotConstant to communicate back to impl lookup that the lookup
+    // failed. This can not happen for a deferred symbolic lookup in a generic
+    // eval block, since we only add the deferred lookup instruction (being
+    // evaluated here) to the semir if the lookup succeeds.
```

Comment:
Maybe it's just me, but I'd always write `SemIR` or (if referring to a specific object) `sem_ir`

```suggestion
    // evaluated here) to the SemIR if the lookup succeeds.
```

Reply:
Done